### PR TITLE
Update from_networkx

### PR DESCRIPTION
**Pull request by PabloAMC**
Changes discussed in 
https://github.com/rusty1s/pytorch_geometric/issues/2920

### CODE REVIEW

File: torch_geometric/utils/convert.py [hunk 1]
```diff
@@ -119,12 +123,23 @@ def from_networkx(G):
...
+    node_attributes = feat_dict.keys()
+    if group_node_attrs is all:
+        group_node_attrs = node_attributes
 
     for i, (_, feat_dict) in enumerate(G.nodes(data=True)):
+        if set(feat_dict.keys()) != set(node_attributes): raise ValueError(f'Not all nodes contain the same attributes')
         for key, value in feat_dict.items():
             data[str(key)] = [value] if i == 0 else data[str(key)] + [value]
 
+    _, _, feat_dict = list(G.edges(data=True))[0]
```
Comment (PabloAMC) [inline torch_geometric/utils/convert.py:136]: I am not sure why this line is failing in the pytest tests, I can run it locally without fail...

Comment (rusty1s) [reply torch_geometric/utils/convert.py:136]: No worries about linting, I will fix that later :)

Comment (rusty1s) [reply torch_geometric/utils/convert.py]: This may be due to different `networkx` versions as they are changing their API frequently.

Comment (PabloAMC) [reply torch_geometric/utils/convert.py]: Actually, it was just for a test of a graph with no edges, and I was looking in the wrong direction. So fortunately that's solved now!

**Comment by PabloAMC** [comment]
I'm not so sure about how to correct the remaining Linting CI fail, but the rest seems to work

**Comment by rusty1s** [comment]
Thanks. Will have a look in the upcoming week.

**Comment by rusty1s** [comment]
Thank you!

**Pull request merged**